gr-dtv/lib/dvbt/dvbt_reference_signals_impl.cc

```diff
@@ -1211,12 +1211,23 @@ dvbt_reference_signals_impl::dvbt_reference_signals_impl(
 {
     d_ninput = ninput;
     d_noutput = noutput;
+    if (config.d_transmission_mode == T2k) {
+        ofdm_fft_size = 2048;
+    } else {
+        ofdm_fft_size = 8192;
+    }
+    ofdm_fft = new (std::nothrow) fft::fft_complex(ofdm_fft_size, false, 1);
+    if (ofdm_fft == NULL) {
+        GR_LOG_FATAL(d_logger, "Reference Signals, cannot allocate memory for ofdm_fft.");
+        throw std::bad_alloc();
+    }
+    normalization = 1.0 / std::sqrt(27.0 * config.d_payload_length);
```

Comment:
might be, but because `config` is initialized in that list also and this uses that variable, I'd say to leave it here ... we don't want initialization ordering requirements in the list (IMHO)!

Reply:
I can't initialize ofdm_fft without using config.d_transmission_mode, so I've moved everything into the initializer list.